src/lib/multicall/Multicall.ts

```diff
@@ -172,7 +179,20 @@ export class Multicall {
       sleep(maxTimeout).then(() => Promise.reject(new Error("multicall timeout"))),
     ]).catch((_viemError) => {
       const e = new Error(_viemError.message.slice(0, 150));
-
+      globalThis.dispatchEvent(
+        new CustomEvent("metrics-mark", {
+          detail: {
+            event: "multicall-timeout",
+            isError: true,
+            message: _viemError.message.slice(0, 150),
+            data: {
+              metricType: "multicall.timeout",
+              isInMainThread: !isWebWorker,
+            } satisfies MulticallMetricData,
+            time: Date.now(),
```

Comment:
time uses for time diffs, not just “time” 
you can send data.timestamp if needed

Reply:
removed